csi/util.go

```diff
@@ -239,6 +242,33 @@ func parseJSONRecurringJobs(jsonRecurringJobs string) ([]longhornclient.Recurrin
 	return recurringJobs, nil
 }
 
+func syncMountPointDirectory(targetPath string) error {
+	d, err := os.OpenFile(targetPath, os.O_SYNC, 0750)
+	if err != nil {
+		return err
+	}
+	if _, err := d.Readdirnames(1); err != nil {
+		if !errors.Is(err, io.EOF) {
+			d.Close()
+			return err
+		}
+	}
+	if err := d.Close(); err != nil {
+		return err
+	}
+
+	// it would not always return `Input/Output Error` or `read-only file system` errors if we only use ReadDir() or Readdirnames() without an I/O operation
+	// an I/O operation will make the targetPath mount point invalid immediately
+	tempFile := targetPath + tempTestMountPointValidStatusFile
+	f, err := os.OpenFile(tempFile, os.O_CREATE|os.O_RDWR|os.O_SYNC|os.O_TRUNC, 0666)
```

Comment:
For the option, although the temp file is invalid, can we just use `O_SYNC` to do the same effect? so don't need to really do a dummy creation?

Reply:
Without `O_CREATE` it will return `no such file or directory` error and following operations will not be done.

Reply:
Thanks to clarify that.